tests/frontend/unit/ExportModal.spec.js

```diff
@@ -0,0 +1,432 @@
+      await wrapper.setProps({
+        view: 'assessment_table'
+      })
+
+      const radioButtons = wrapper.findAll('[datacy^="exportModal:docxExportType"]')
+      expect(radioButtons.length).toBeGreaterThan(0)
+    })
+
+    it('displays radio buttons for selecting the structuring of the docx export if docx.exportTypes option is true, the selected docx template is "condensed", and the view is "assessment_table"', async () => {
+      await wrapper.setProps({
+        view: 'assessment_table'
+      })
+      const radioButtonStubs = wrapper.findAll('[datacy^="exportModal:docxSortType"]')
+
+      expect(radioButtonStubs.length).toBeGreaterThan(0)
+    })
+
+    it('displays an explanation if docx.anonymize and docx.obscure options are false and docx.exportTypes and docx.templates options are false', async () => {
+      await wrapper.setProps({
+        options: {
+          ...props.options,
+          docx: {
+            ...props.options.docx,
+            anonymize: false,
+            exportTypes: false,
+            obscure: false,
+            templates: false
+          }
+        }
+      })
+      const explanationText = 'explanation.export.anonymous'
+
+      expect(wrapper.html()).toContain(explanationText)
+    })
+
+    it('displays a message that docx fragment export is deactivated if the view mode is not "view_mode_default"', async () => {
+      await wrapper.setProps({
+        viewMode: 'view_mode_tag'
+      })
+      const message = 'explanation.export.disabled.viewMode'
+      const docxDiv = wrapper.find('#docx')
+
+      expect(docxDiv.html()).toContain(message)
+    })
+
+    it('does not display a message that docx fragment export is deactivated if the view mode is "view_mode_default"', () => {
+      const message = 'explanation.export.disabled.viewMode'
+      const docxDiv = wrapper.find('#docx')
+
+      expect(docxDiv.html()).not.toContain(message)
+    })
+  })
+
+  describe('ExportModal: xlsx export', () => {
+    it('displays "anonymized" checkbox in the xlsx tab if xlsx.anonymize or xlsx.obscure option is set to true', async () => {
+      await wrapper.setProps({
+        options: {
+          ...props.options,
+          xlsx: {
+            ...props.options.xlsx,
+            anonymize: true
+          }
+        }
+      })
+      const checkboxStub = wrapper.find('[datacy="exportModal:xlsxAnonymous"]')
+
+      expect(checkboxStub.exists()).toBe(true)
+    })
+
+    it('does not display "anonymized" checkbox in the xlsx tab if xlsx.anonymize and xlsx.obscure options are set to false',  () => {
+      const checkboxStub = wrapper.find('[datacy="exportModal:xlsxAnonymous"]')
+
+      expect(checkboxStub.exists()).toBe(false)
+    })
+
+    // fails because 'feature_admin_assessmenttable_export_statement_generic_xlsx' permission is always evaluated as true
```

Comment:
Shouldn't we rather remove this and craft another test which doesn't require the evaluation of `feature_admin_assessmenttable_export_statement_generic_xlsx`?

Reply:
I'm not sure how? Right now, all permission checks evaluate to true in our unit tests because of
`const hasPermission = jest.fn(() => true)`
in VueConfigLocal.js. This means there will always be the third radio button (which is rendered if `feature_admin_assessmenttable_export_statement_generic_xlsx` is true), so this test cannot succeed.
Of course we should fix permissions in our unit tests asap, but I suggest doing it in a separate PR.

Reply:
Got it, thanks for the explanations :+1: